firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Driver/I2C/SFM3019/Device.txt

```diff
@@ -0,0 +1,253 @@
+      THEN('The mock I2C's read buffer remains empty') 
+      THEN('The global_device I2C's read buffer remains empty') 
+      THEN('The global_device I2C's write buffer remains unchanged') 
+
+    WHEN('The set_averaging function is called wirh averaging_window 128 as input parameter') 
+      THEN('The set_averaging function returns ok status') 
+      THEN('The mock I2C's write buffer consists of set_averaging  command byte '0x366A' followed by '0x0080FB') 
+      THEN('The mock I2C's read buffer remains empty') 
+      THEN('The global_device I2C's read buffer remains empty') 
+      THEN('The global_device I2C's write buffer remains unchanged ') 
+
+Scenario: SFM3019 Device:: SFM3019 device behaves correctly when request_conversion_factors function is called 
+  GIVEN('A SFM3019 device constructed with a mock I2C device , global I2C device has write buffer '0x01' and gas parameter') 
+    WHEN('The request_conversion_factors function is called with gas type O2 as input parameter') 
+      THEN('The request_conversion_factors function returns ok status') 
+      THEN('The mock I2C's write buffer consists of request_conversion_factor O2 command bytes '0x3661 followed by '0x36033A'') 
+      THEN('The mock I2C's read buffer remains empty') 
+      THEN('The global_device I2C's read buffer remains empty') 
+      THEN('The global_device I2C's write buffer remains unchanged ') 
+
+    WHEN('The request_conversion_factors function is called with  gas type air as input parameter') 
+      THEN('The request_conversion_factors function returns ok status') 
+      THEN('The mock I2C's write buffer consists of request_conversion_factor air command bytes '0x3661 followed by '0x3608D0'') 
+      THEN('The mock I2C's read buffer remains empty') 
+      THEN('The global_device I2C's read buffer remains empty') 
+      THEN('The global_device I2C's write buffer remains unchanged') 
+
+    WHEN(
+        'The request_conversion_factors function is called with gas type mixture as input parameter') 
+      THEN('The request_conversion_factors returns ok') 
+      THEN('The mock I2C's write buffer consists of request_conversion_factor mixture command byte '0x3661 followed by '0x3632CE'') 
+      THEN('The mock I2C's read buffer remains empty') 
+      THEN('The global_device I2C's read buffer remains empty') 
+      THEN('The global_device I2C's write buffer remains unchanged') 
+
+Scenario: SFM3019 Device:: SFM3019 device behaves correctly when read_conversion_factors function is called 
+  GIVEN('A SFM3019 device constructed with a mock I2C device having read buffer 0x[ae dc 6a 00 00 81 00 00 81], global I2C device has write buffer '0x01' and air as gas parameter') 
+    WHEN(The read_conversion_factors function is called with a default-initialized ConversionFactors output parameter and with the read buffer of mock I2C device consisting 0x[01 02], corresponding to scale factor 10 and offset of -5') 
```

Comment:
All contents of the read buffer (both the stuff we want to read, and the "junk bytes" afterwards) should be specified in this WHEN: WHEN we call ... with read buffer consisting of bytes ae dc ... 81 (corresponding to command byte of 0x1234, offset of -5 and then scale factor of 10) and 01 02 (as junk bytes)